app/src/main/java/com/tolkiana/tacoapp/TacoApplication.kt

```diff
@@ -0,0 +1,22 @@
+package com.tolkiana.tacoapp
+
+import android.app.Application
+import com.android.volley.toolbox.ImageLoader
+import com.android.volley.toolbox.Volley
+import com.tolkiana.tacoapp.utilities.BitmapCache
+import com.tolkiana.tacoapp.utilities.DataParser
+
+/**
+ * Created by tolkiana on 9/10/17.
+ */
+class TacoApplication: Application() {
+
+    var imageLoader: ImageLoader? = null
+    var dataParser: DataParser? = null
```

Comment:
Instead of creating a `DataParser` at application initialization and referencing it from the application class, you can alternatively use Kotlin's [object](https://kotlinlang.org/docs/reference/object-declarations.html#object-declarations) keyword to create a Singleton. For example, the `DataParser` could look like this:

```kt
object DataParser {
    val mapper = jacksonObjectMapper()

    inline fun <reified T> parseArrayFromAsset(assetManager: AssetManager, fileName: String): List<T> {
        val asset = assetManager.fileAsString(fileName)
        val collectionType = mapper.typeFactory.constructCollectionType(List::class.java, T::class.java)
        return mapper.readValue(asset, collectionType)
    }

    inline fun <reified T> parseObjectFromAsset(assetManager: AssetManager, fileName: String): T {
        val asset = assetManager.fileAsString(fileName)
        return mapper.readValue(asset, T::class.java)
    }

    fun AssetManager.fileAsString(filename: String): String {
        return open(filename).use {
            it.readBytes().toString(Charset.defaultCharset())
        }
    }
}
```

The `DataParser` object class will be initialized when it is first used in your `TacoListActivity`:

```kt
val tacoList = DataParser.parseArrayFromAsset<Product>(this.assets, "tacos.json")
```

Kotlin makes it really convenient to work with Singletons that don't require constructor parameters, and I think what you're after by accessing `dataParser` from the application class could be more Kotlin-like by using the `object` keyword to create a Singleton.